browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -53,6 +55,21 @@ RewardsTabHelper::~RewardsTabHelper() {
 #endif
 }
 
+void RewardsTabHelper::BindCreatorDetectionHost(
+    mojo::PendingAssociatedReceiver<mojom::CreatorDetectionHost> receiver,
+    content::RenderFrameHost* rfh) {
+  auto* web_contents = content::WebContents::FromRenderFrameHost(rfh);
+  if (!web_contents) {
+    return;
+  }
+  auto* tab_helper = RewardsTabHelper::FromWebContents(web_contents);
+  if (!tab_helper) {
+    return;
+  }
+  tab_helper->receiver_.reset();
```

Comment:
Here you usually have 1 (tab helper) - many (frame) relationship.
Even for you process only main frames (taking into account BF cache).

Reply:
Updated to use `RenderFrameHostReceiverSet`.